system/jlib/jmutex.hpp

```diff
@@ -373,21 +375,18 @@ class  SpinLock
 
 class jlib_decl  SpinLock
 {
-    atomic_t value;
-    unsigned nesting;           // not volatile since it is only accessed by one thread at a time
-    struct { volatile ThreadId tid; } owner;
-    inline SpinLock(SpinLock & value __attribute__((unused))) { assert(false); } // dummy to prevent inadvetant use as block
+    std::atomic_uint value{false};  // Use an atomic_uint rather than a bool because it is more efficient on power8
+    unsigned nesting = 0;           // This is not atomic since it is only accessed by one thread at a time
+    std::atomic<ThreadId> owner{0};
+    inline SpinLock(SpinLock & value __attribute__((unused))) = delete; // to prevent inadvertent use as block
 public:
-    inline SpinLock()       
+    inline SpinLock()
     {   
-        owner.tid = 0;
-        nesting = 0;
-        atomic_set(&value, 0); 
     }
 #ifdef _DEBUG
     ~SpinLock()             
     { 
-        if (atomic_read(&value))
+        if (value)
```

Comment:
trivial - why value and not (just to be consistent) value.load(std::memory_order_xxx) ?